bit_manipulation/bitmask.cpp

```diff
@@ -0,0 +1,167 @@
+/**
+ * @file
+ * @brief Bitmask example
+ * @details
+ * (Bitmask)[https://blog.bitsrc.io/the-art-of-bitmasking-ec58ab1b4c03] 
+ * example with explanation and with example
+ * problem solution using bitmask
+ * 
+ * @author [Yhtyyar Sahatov](https://github.com/Yhtiyar)
+ */
+
+#include <iostream>   /// for IO operations
+/**
+ * @namespace bitmask
+ * @brief Bitmask examples
+ */ 
+namespace bitmask 
+{
+/**
+ * This function returns the bit representation of mask(number) in given length .
+ * @param mask the mask
+ * @param n lenght of mask
+ * @returns returns the bit representation of mask(number) in given length as string.
+ */
+std::string to_string(int mask, int n);
+
+/**
+ * This function demonstrates basics of bitmask.
```

Comment:
When there's no `@param` text in it, we need to add `@brief` and the return type, otherwise, Doxygen won't recognize this function.
```suggestion
 * @brief Demonstrates the basics of Bitmask
 * @returns void
```